### PR TITLE
WIP: Enable profiling in jemalloc.

### PR DESCRIPTION
**Pull request by mberhault**
Also switch over to **restrict** keyword and link in math (needed for
log() in prof.c)

### CODE REVIEW

File: cgo_flags.go [hunk 1]
```diff
@@ -6,4 +6,5 @@ package jemalloc
 // #cgo linux CFLAGS: -DJEMALLOC_PURGE_MADVISE_DONTNEED
 // #cgo darwin CFLAGS: -DJEMALLOC_OSATOMIC -DJEMALLOC_OSSPIN -DJEMALLOC_PURGE_MADVISE_FREE -DJEMALLOC_HAVE_ISSETUGID
 // #cgo darwin CFLAGS: -DJEMALLOC_ZONE -DJEMALLOC_ZONE_VERSION=8
+// #cgo LDFLAGS: -Wl,-undefined -Wl,dynamic_lookup -lm
```
Comment (bdarnell) [inline cgo_flags.go:9]: Have you tested this on both linux and OSX? All our other `-Wl` flags need both darwin and non-drawin variants.

File: internal/include/jemalloc/internal/util.h [hunk 2]
```diff
@@ -140,8 +140,8 @@
 #ifdef JEMALLOC_H_EXTERNS
 
 int	buferror(int err, char *buf, size_t buflen);
-uintmax_t	malloc_strtoumax(const char *restrict nptr,
-    char **restrict endptr, int base);
+uintmax_t	malloc_strtoumax(const char* __restrict__ nptr,
```
Comment (bdarnell) [inline internal/include/jemalloc/internal/util.h:143]: Should we be unsetting JEMALLOC_HAS_RESTRICT in jemalloc_internal_defs.h instead?

File: cgo_flags.go (hunk 1 above)
Comment (mberhault) [inline cgo_flags.go:9]: not yet. I'm trying to get it basic profiling to work on linux first. Once that's done, I'll worry about making sure it's all neat and clean, including build flags and various DEFINES

Comment (bdarnell) [inline cgo_flags.go:9]: Nevermind, missed the `WIP` in the subject. But oddly this line is used for darwin in c-rocksdb, not linux.

**Comment by tamird** [comment]
> Also switch over to restrict keyword

What does this do? FYI this is changing code from upstream, and is likely to be lost in subsequent updates.

**Comment by mberhault** [comment]
that's just to get it to build for now. as Ben pointed out, changing a define might be enough, otherwise, rerunning configure may do the trick as well.

**Comment by tamird** [comment]
I think this is superseded by #2. Close?

**Comment by mberhault** [comment]
yup. closing.

**Pull request closed without merging**